### ISSUE TITLE
NodeJs: TypeError: window.setTimeout is not a function

### ISSUE THREAD

**Issue by minaEweida** [badoo/Reaktive#485]
I recently started supporting Js on a library that uses reaktive and I had to publish it as an npm package and part of this process involved exporting all dependencies including Reaktive.
I was able to locally publish everything including Reaktive and initialize my library on a sample nodejs app but I got an error on my sample app:

`ReferenceError: window is not defined`

Because of this line (This is the javascript code that was publish not kotlin)
`var element = window.setTimeout(task, delayMillis.toInt());`

Which is actually in this function in `MainScheduler.kt` 
```
        override fun submit(delayMillis: Long, task: () -> Unit) {
            timeoutIds += window.setTimeout(task, delayMillis.toInt())
        }
```
I worked around it by doing this

```
global.window = {document: {createElementNS: () => {return {}} }};
```

But I got another error and I got stuck with it
```
TypeError: window.setTimeout is not a function
```

I am not nodejs expert so I wanted to ask, should the implementation work fine and the issue is from my side or the line below should be different in case of nodeJs not browser. Because I am assuming this should work fine on a browser? I would also assume `window.setInterval` should break for me as well.

```
timeoutIds += window.setTimeout(task, delayMillis.toInt())
```

**Comment by arkivanov** [comment]
@minaEweida Thanks for reporting, this looks like a bug.

**Comment by arkivanov** [comment]
@lukaville Would you like to take this?

**Comment by minaEweida** [comment]
@arkivanov I did a fix that worked for me locally, I can share later today if it's not going to be fixed soon. Will send for review and then cleanup.

**Comment by arkivanov** [comment]
@minaEweida Thanks, PR is very welcome!

**Comment by arkivanov** [comment]
Closed by #486
